packages/aws-cdk/lib/api/deployments/asset-publishing.ts

```diff
@@ -184,7 +184,7 @@ export const EVENT_TO_LEVEL: Record<EventType, IoMessageLevel | false> = {
 
 export abstract class BasePublishProgressListener implements IPublishProgressListener {
   protected readonly ioHost: IIoHost;
-  protected readonly action: ToolkitAction;
+  protected readonly action: IoMessaging['action'];
```

Comment:
Plan is to not have the `action` here but to have an `ActionAwareIoHost`. See `importer.ts` as an example. Eventually we will just be passing `ActionAwareIoHost` hosts around.